### ISSUE TITLE
Playing variations and skipping to next comments can mess up colors

### ISSUE THREAD

**Issue by sphaso** [neagle/gokibitz#97]
Heya, I found another bug in the dreaded Lumo vs. execration game.
Steps:
- go to https://gokibitz.com/kifu/VJdTzp84?path=60-58:1
- play the very first sequence Ben provides
- skip to the next comment (with the comment arrow >)
- the stone at Q7 is actually black! you can verify this by replaying the steps without 2 (playing the sequence)

This is the only instance I found, but I'm on the lookout :)

**Comment by levelonedev** [comment]
I'm not seeing what you are talking about. It's a different variation. If you look at the query string it is saying that you are on move 60 but branched from move 58 on variation 1 (0 based variation labelling so this is variation 2).  When you click the next comment button you are now on ?path=59-58:1-59:1 Move 59 branches from variation 1 on move 58 and now this is another branch in that variation. (If I am wrong please help me see the change).

But there is definitely a problem with navigating with the next/previous comment buttons. I can't move to a previous move before the one you link!

**Comment by sphaso** [comment]
I know it's a different variation, but the stone is black regardless :)
Screenshots:
1) go to https://gokibitz.com/kifu/VJdTzp84?path=60-58:1
![screenshot from 2015-06-02 07-37-57](https://cloud.githubusercontent.com/assets/4958220/7929148/f4f37ce2-08fa-11e5-8adb-035717dd1c9c.png)
2) play the very first sequence Ben provides
![screenshot from 2015-06-02 07-38-29](https://cloud.githubusercontent.com/assets/4958220/7929154/0810ce60-08fb-11e5-8389-8ec588700df9.png)
3) skip to the next comment (with the comment arrow >)
4) the stone at Q7 is actually black! *It shouldn't be white*
![screenshot from 2015-06-02 07-38-47](https://cloud.githubusercontent.com/assets/4958220/7929162/217304fe-08fb-11e5-8029-7eaeae236430.png)
5) evidence? If I refresh the page it becomes black :)
![screenshot from 2015-06-02 07-39-03](https://cloud.githubusercontent.com/assets/4958220/7929169/309a9faa-08fb-11e5-8151-1adbf998722b.png)

**Comment by levelonedev** [comment]
What browser are you using because it appears the correct color for me.

**Comment by sphaso** [comment]
Oh, interesting. This is Firefox 38.0.1 on Arch Linux.
I've just noticed that as soon as I get to the URL I get a console.log:
"Error: null" app.js:3:17496

**Comment by levelonedev** [comment]
I'm using Firefox 38.0 on Ubuntu 14.04. This is so odd.

**Comment by sphaso** [comment]
Bored at work. Can reproduce on a Windows machine on:
- Firefox 38.0.5
- Chrome 43.0.2357.81 m
- Vivaldi 1.0.118.19
- IE 10

**Comment by neagle** [comment]
I totally believe that this is happening... basically, the initial way I implemented sequences is kind of inadequate. It has one major failing (stones aren't captured because the stones and numbers are just added as _labels_ to WGo) and some minor ones (there are some fringe cases of disappearing or stuck stones). The way this needs to be fixed is by a reimplementation of the feature. Which is high priority for me, but I'm just not getting the time I wish I had to work on features for GoKibitz right now.

Basically, I need to get a) password change + email verification at signup working, and then this is probably my next priority.
